csrc/velox/functions/text/vocab_ops.h

```diff
@@ -0,0 +1,26 @@
+
+#pragma once
+
+#include <vector>
+#include "velox/functions/Udf.h"
+#include "velox/type/Type.h"
+#include "vocab.h"
+
+namespace facebook::torcharrow::functions {
+template <typename T> 
+struct lookup_indices {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE bool
+  call(velox::exec::ArrayWriter<int64_t> &result,
+       const arg_type<std::shared_ptr<Vocab>> &vocab,
+       const arg_type<velox::Array<velox::Varchar>> &tokens) {
+    std::vector<std::string> input;
+    for(const auto& token:tokens){
+	    std::string token_str = token.str();
```

Comment:
@wenleix I am hitting compilation issue here. 

```
error: no member named 'str' in 'facebook::velox::exec::VectorOptionalValueAccessor<facebook::velox::exec::VectorReader<facebook::velox::Varchar>>'
            std::string token_str = token.str();
```

Any suggestions on what's wrong here?

Reply:
Could you try adding `#include <string>` to the top of the file?

Reply:
> Could you try adding `#include <string>` to the top of the file?

Yupp, same issue with include. What surprise me is that .str() on velox::Varchar object works on your PR [here](https://github.com/facebookresearch/torcharrow/pull/271/files#diff-46806fbc330a7928043d248e630f4f21df4a126c985a1d17bd8d37a069c65198R30).

Reply:
Seem like calling `.value` get rid of compilation issue. Got some idea from here https://github.com/facebookincubator/velox/blob/2434e487eb96e8cf45a1949e971ad2d7022eabca/velox/functions/prestosql/ArrayFunctions.h#L98 although I don't fully grasp the underlying mechanics yet.